src/mscorlib/src/System/Buffer.cs

```diff
@@ -428,15 +431,275 @@ internal unsafe static void Memmove(byte* dest, byte* src, nuint len)
             PInvoke:
             _Memmove(dest, src, len);
         }
-        
-        // Non-inlinable wrapper around the QCall that avoids poluting the fast path
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void Memmove<T>(ref T destination, ref T source, int elementCount)
```

Comment:
I would avoid this intermediate layer and moved the nuint -> int cast to the caller.

Note that the JIT is not able to inline generic methods called from other generic methods in shared generic code today. This will avoid some of the problem.

Reply:
Addressed in https://github.com/dotnet/coreclr/pull/15947/commits/39abfd0b2625c60df33e1d0d294ec64ddb279b9d.